src/cpu/flexcpu/inflight_inst.hh

```diff
@@ -0,0 +1,592 @@
+                           const LaneData<LaneSize::Byte>& val) override;
+    void setVecLaneOperand(const StaticInst* si, int dst_idx,
+                           const LaneData<LaneSize::TwoByte>& val) override;
+    void setVecLaneOperand(const StaticInst* si, int dst_idx,
+                           const LaneData<LaneSize::FourByte>& val) override;
+    void setVecLaneOperand(const StaticInst* si, int dst_idx,
+                           const LaneData<LaneSize::EightByte>& val) override;
+
+    VecElem readVecElemOperand(const StaticInst* si,
+                               int op_idx) const override;
+    void setVecElemOperand(const StaticInst* si, int dst_idx,
+                           const VecElem val) override;
+
+    const VecPredRegContainer&
+    readVecPredRegOperand(const StaticInst *si, int idx) const override;
+
+    VecPredRegContainer&
+    getWritableVecPredRegOperand(const StaticInst *si, int idx) override;
+
+    void
+    setVecPredRegOperand(const StaticInst *si, int idx,
+                         const VecPredRegContainer& val) override;
+
+    RegVal readCCRegOperand(const StaticInst* si, int op_idx) override;
+    void setCCRegOperand(const StaticInst* si, int dst_idx, RegVal val)
+                         override;
+
+    RegVal readMiscRegOperand(const StaticInst* si, int op_idx) override;
+    void setMiscRegOperand(const StaticInst* si, int dst_idx,
+                           RegVal val) override;
+
+    RegVal readMiscReg(int misc_reg) override;
+    void setMiscReg(int misc_reg, RegVal val) override;
+
+    PCState pcState() const override;
+    void pcState(const PCState& val) override;
+
+
+    Fault readMem(Addr addr, uint8_t *data, unsigned int size,
+                  Request::Flags flags,
+                  const std::vector<bool>& byte_enable = {}) override;
+
+    Fault initiateMemRead(Addr addr, unsigned int size, Request::Flags flags,
+                          const std::vector<bool>& byte_enable = {}) override;
+
+    Fault writeMem(uint8_t* data, unsigned int size, Addr addr,
+                   Request::Flags flags, uint64_t* res,
+                   const std::vector<bool>& byte_enable = {}) override;
+
+
+    void setStCondFailures(unsigned int sc_failures) override;
+
+    unsigned int readStCondFailures() const override;
+
+    void syscall(int64_t callnum, Fault* fault) override;
+
+    ThreadContext* tcBase() override;
+
+    /**
+     * ARM-specific
+     */
+    bool readPredicate() const override;
+    void setPredicate(bool val) override;
+    bool readMemAccPredicate() const override;
+    void setMemAccPredicate(bool val) override;
+
+    /**
+     * X86-specific
+     */
+    void demapPage(Addr vaddr, uint64_t asn) override;
+    void armMonitor(Addr address) override;
+    bool mwait(PacketPtr pkt) override;
+    void mwaitAtomic(ThreadContext* tc) override;
+    AddressMonitor* getAddrMonitor() override;
+
+    /**
+     * MIPS-Specific
+     */
+#if THE_ISA == MIPS_ISA
```

Comment:
You can drop this, if you want, and just say you don't support MIPS. It might be safer that way, since it's untested.

Reply:
I had it this way because the ExecContext has these lines. It's there so that it would compile if the CPU was included in MIPS, and still potentially run if these interface functions were not called. It would be nice for someone to fully implement things at some point, so I don't know if it's better to actually remove it.